perftestnotebook/analyzer.py

```diff
@@ -41,8 +41,23 @@ def get_header(self):
             template_header_content = f.read()
             return template_header_content
 
-    def get_notebook_section(self, func):
+    def get_notebook_section(self, function_name,function_parameter=None):
+        notebook_section = ""
         template_function_folder_path = "testing/resources/notebook-sections/"
-        template_function_file_path = os.path.join(template_function_folder_path, func)
+        template_function_file_path = os.path.join(template_function_folder_path, function_name)
         with open(template_function_file_path, "r") as f:
-            return f.read()
+            line = f.readline()
```

Comment:
Is there a reason you are using `readline` here? You can use `readlines` to get all lines and iterate over those in a for loop.

Reply:
I thought readline would be faster. readlines would pass the file 1 time, then traverse the list another time.

Reply:
> Is there a reason you are using `readline` here? You can use `readlines` to get all lines and iterate over those in a for loop.

I thought readline would be faster. readlines would pass the file 1 time, then traverse the list another time.

Reply:
I agree, it's probably a little bit faster, but it's nothing noticeable here - the lists are pretty small.